modules/frontend/search_sharder.go

```diff
@@ -225,13 +225,46 @@ func (s *asyncSearchSharder) ingesterRequest(ctx context.Context, tenantID strin
 
 	// if ingester start == ingester end then we don't need to query it
 	if ingesterStart == ingesterEnd {
-		return nil, nil
+		return nil
 	}
 
 	searchReq.Start = ingesterStart
 	searchReq.End = ingesterEnd
 
-	return buildIngesterRequest(ctx, tenantID, parent, &searchReq)
+	// Split the start and end range into sub requests for each range.
+	duration := searchReq.End - searchReq.Start
+	interval := duration / uint32(s.cfg.IngesterShards)
```

Comment:
i think this will underscan by a few seconds due to the flooring of integer division.

maybe make the break condition the for loop `shardStart > ingesterEnd` instead of ingester shards? a lot of fixes for this.

Reply:
Thanks for calling this out.  The tests have been updated and verified against this case.